pkgs/applications/radio/gnuradio/wrapper.nix

```diff
@@ -28,6 +28,27 @@
 , extraPackages ? []
 # For Adding additional python packaages
 , extraPythonPackages ? []
+, soapysdr # For it's passthru.searchPath
+# soapysdr plugins we add by default. Ideally, we should have a
+# soapysdrPackages = soapysdr.pkgs attribute set, but until now this wasn't
+# crucial.
+, soapyairspy
```

Comment:
The duplication here is really not so nice, also if there was a way to have the logic about which wrapper arguments are needed inside of the soapy expression that would be somewhat nicer and allow other dependencies to also not have to rebuild when changing the available plugins. Wouldn't know how that would look like in practice though.

Reply:
> The duplication here is really not so nice

You mean, the duplication of the soapy packages as arguments, and also them in the `extraSoapyPackages ? [ ... ]` list?

> also if there was a way to have the logic about which wrapper arguments are needed inside of the soapy expression

Do you mean the wrapper in `soapysdr-with-plugins`?

> allow other dependencies to also not have to rebuild when changing the available plugins

I'm sorry but I'm not sure I understand..

Reply:
I was thinking (before it was merged without time to disapprove), that maybe `gnuradio.unwrapped` should inherit `soapysdr` in it's `passthru`, and then use that in the wrapper. Then, in case you override `soapysdr`, you will also get the wrapper use the overriden `soapysdr`. However, because we lack `soapysdr.pkgs` attribute set, you will anyway have to override each of these `soapysdr` plugins to use the overriden `soapysdr`...

Currently, if you want to override `soapysdr` for Gnuradio, the safest thing would be to use `overrideScope`.